lib/core/request.c

```diff
@@ -71,6 +71,12 @@ static struct st_deferred_request_action_t *create_deferred_action(h2o_req_t *re
     return action;
 }
 
+static void strtolower_cpy(char *d, const char *s, size_t len)
+{
+    for (; len != 0; ++d, ++s, --len)
+        *d = h2o_tolower(*s);
+}
```

Comment:
How about doing the following?
* Promote this function to an external function that sits next to `h2o_strtolower`, and maybe name it `h2o_strcopytolower`?
* Let `h2o_strtolower` be a wrapper of `h2o_strcopytolower` that accepts the same address for `d` and `s`.

The intent here is to avoid having duplicated code while keeping existing API consistent.

Reply:
Sounds good, thank you!